pkg/reconciler/chart/factory.go

```diff
@@ -235,13 +235,31 @@ func (f *DefaultFactory) downloadComponent(component *Component, dstDir string)
 	return nil
 }
 
-func (f *DefaultFactory) downloadArchive(URL, dstDir string) (string, error) {
-	f.logger.Infof("Downloading archive '%s' into workspace '%s'", URL, dstDir)
+func (f *DefaultFactory) downloadArchive(URL, dstDir string, authenticator ExternalComponentAuthenticator) (string, error) {
+
+	req, err := http.NewRequest("GET", URL, nil)
 
-	resp, err := http.Get(URL) // #nosec
+	if err != nil {
+		return "", errors.New("cannot create request")
+	}
+
+	if authenticator != nil {
+		f.logger.Infof("Downloading archive '%s' into workspace '%s' from private repo", URL, dstDir)
+
+		err = authenticator.Do(req)
```

Comment:
authenticator.Do(req) always returns nil. We don't need to process it. We can also get rid of returned error value from signature

`(e ExternalComponentAuthenticator) Do(r *http.Request) error `

Reply:
Done